recipes-kernel/kernel-tests/kernel-test-nohz.bb

```diff
@@ -31,4 +31,20 @@ do_install_ptest_append() {
     install -m 0755 ${S}/nohz_test ${D}${PTEST_PATH}
 }
 
+pkg_postinst_ontarget_${PN}-ptest_append() {
+    CPUS=`nproc`
+    ISOLATED_CPU=$((CPUS - 1))
+
+    if [ $ISOLATED_CPU -gt 0 ]; then
+        fw_setenv othbootargs 'isolcpus='$ISOLATED_CPU' nohz_full='$ISOLATED_CPU
```

Comment:
Does this imply that the ptest requires a reboot between package installation and executing the test?

In that case, I think it would be more appropriate to include a tool with the ptest subpackage which the user can intentionally run and insert these options into their cmdline (or run-in-reverse to remove them). Setting them at package-install time seems inappropriate to me.

Reply:
Yes, this test requires a reboot; however:

- changing the kernel command line is not the only reason a reboot is required for this test. The main reason is it depends on and installs a new kernel (configured for NO_HZ_FULL). Basically even w/o these postinst/postrm scripts a reboot will be required.
- the ATS this is going to be run in already has a reboot in place after the install.

I would prefer to keep this as a postinst script because it greatly simplifies the main use case: running this ptest as part of the ATS. With the current implementation all the ATS needs to do is install this ptest, reboot and run ptest-runner. If I start to introduce custom scripts that means the ptest is no longer self contained and the ATS framework now needs to know that for this particular ptest additional scripts need to be called (to change the command line). This adds unneeded complication.

For somebody running this test by hand I can print a message advising about a reboot being required. I expect manually running this test to be a corner case that's not encountered very often outside the ATS.